### PR TITLE
transactional events in subscriber interface

### PR DESCRIPTION
**Pull request by pleerock**

### CODE REVIEW

**Comment by pleerock** [comment]
@imnotjames I'm going to merge this one and release a new version. Do you want to review it before I do that?

File: src/find-options/FindOneOptions.ts [hunk 1]
```diff
@@ -59,4 +59,9 @@ export interface FindOneOptions<Entity = any> {
      */
     loadEagerRelations?: boolean;
 
+    /**
+     * If this is set to true, SELECT query in a `find` method will be executed in a transaction.
+     */
+    transaction?: boolean
```
Comment (imnotjames) [inline src/find-options/FindOneOptions.ts:65]: I'm assuming this was intentionally added but seems to be a separate feature from the rest of this PR

**Review by imnotjames — APPROVED** [review]
Overall makes sense.

This does make the disparity between listeners, schema listeners, and subscribers wider, but there's already disparity there as it stands.

File: src/find-options/FindOneOptions.ts (hunk 1 above)
Comment (pleerock) [reply src/find-options/FindOneOptions.ts:65]: oh yeah I forgot I need to create a separate branch for this separate feature

**Comment by pleerock** [comment]
> This does make the disparity between listeners, schema listeners, and subscribers wider, but there's already disparity there as it stands.

yes it does, and here is why:

when you run:

```sql
START TRANSACTION
SELECT * FROM user
COMMIT
```

you execute:

```ts
entity.beforeTransactionStart()
entity.afterTransactionStart()
```

there is no "entity" before SELECT on which we can call `beforeTransactionStart` method.

**Pull request merged**